src/Datadog.Trace.ClrProfiler.Managed/Integrations/GraphQLIntegration.cs

```diff
@@ -0,0 +1,351 @@
+        [InterceptMethod(
+            TargetAssembly = GraphQLAssemblyName,
+            TargetType = GraphQLDocumentValidatorInterfaceName,
+            TargetSignatureTypes = new[] { GraphQLValidationResultInterfaceName, ClrNames.Ignore, ClrNames.Ignore, ClrNames.Ignore, ClrNames.Ignore, ClrNames.Ignore, ClrNames.Ignore })]
+        public static object Validate(
+            object documentValidator,
+            object originalQuery,
+            object schema,
+            object document,
+            object rules,
+            object userContext,
+            object inputs,
+            int opCode,
+            int mdToken)
+        {
+            if (documentValidator == null) { throw new ArgumentNullException(nameof(documentValidator)); }
+
+            const string methodName = nameof(Validate);
+
+            // At runtime, get a Type object for GraphQL.ExecutionResult
+            var documentValidatorInstanceType = documentValidator.GetType();
+            Type graphQLExecutionResultType;
+            Type documentValidatorInterfaceType;
+
+            try
+            {
+                var graphQLAssembly = AppDomain.CurrentDomain.GetAssemblies()
+                                        .Where(a => a.GetName().Name.Equals(GraphQLAssemblyName))
+                                        .Single();
+                graphQLExecutionResultType = graphQLAssembly.GetType(GraphQLExecutionResultName, throwOnError: true);
+                documentValidatorInterfaceType = graphQLAssembly.GetType(GraphQLDocumentValidatorInterfaceName, throwOnError: true);
+            }
+            catch (Exception ex)
+            {
+                // This shouldn't happen because the GraphQL assembly should have been loaded to construct various other types
+                // profiled app will not continue working as expected without this method
+                Log.ErrorException($"Error calling {documentValidatorInstanceType.Name}.{methodName}(IConnection connection, CancellationToken cancellationToken)", ex);
+                throw;
+            }
+
+            Func<object, object, object, object, object, object, object, object> instrumentedMethod;
+
+            try
+            {
+                instrumentedMethod =
+                    MethodBuilder<Func<object, object, object, object, object, object, object, object>>
+                        .Start(Assembly.GetCallingAssembly(), mdToken, opCode, methodName)
+                        .WithConcreteType(documentValidatorInstanceType)
+                        .WithParameters(originalQuery, schema, document, rules, userContext, inputs)
+                        .Build();
+            }
+            catch (Exception ex)
+            {
+                // profiled app will not continue working as expected without this method
+                Log.ErrorException($"Error resolving {documentValidatorInstanceType.Name}.{methodName}(IConnection connection, CancellationToken cancellationToken)", ex);
```

Comment:
Are the parameters a copy paste miss?

Reply:
Yep, fixed these statements in bec643fc158484cc1a8b85e459fddc7aac654959